packages/platforms/delta/armel/arm-delta-ag6248c/src/arm_delta_ag6248c/module/src/sysi.c

```diff
@@ -0,0 +1,290 @@
+#include <onlplib/file.h>
+#include <onlp/platformi/sysi.h>
+#include <onlp/platformi/ledi.h>
+#include <onlp/platformi/thermali.h>
+#include <onlp/platformi/fani.h>
+#include <onlp/platformi/psui.h>
+
+#include "arm_delta_ag6248c_int.h"
+#include "arm_delta_ag6248c_log.h"
+
+#include "platform_lib.h"
+#include "arm_delta_i2c.h"
+platform_id_t platform_id = PLATFORM_ID_UNKNOWN;
+
+#define ONIE_PLATFORM_NAME "arm-delta-ag6248c-r0"
+
+const char*
+onlp_sysi_platform_get(void)
+{ 
+	enum ag6248c_product_id pid = get_product_id();
+	
+	if (pid == PID_AG6248C_48)
+		return "arm-delta-ag6248c";
+    else if(pid == PID_AG6248C_48P)
+		return "arm-delta-ag6248c-poe";
+	else 
+		return "unknow";
+}
+
+int
+onlp_sysi_platform_set(const char* platform)
+{ 
+    if(strstr(platform,"arm-delta-ag6248c-r0")) {
+        platform_id = PLATFORM_ID_POWERPC_DELTA_AG6248C_R0;
+        return ONLP_STATUS_OK;
+    }
+    if(strstr(platform,"arm-delta-ag6248c-poe-r0")) {
+        platform_id = PLATFORM_ID_POWERPC_DELTA_AG6248C_POE_R0;
+        return ONLP_STATUS_OK;
+    }
+	AIM_LOG_ERROR("No support for platform '%s'", platform);
+    return ONLP_STATUS_E_UNSUPPORTED;
+}
+
+int
+onlp_sysi_platform_info_get(onlp_platform_info_t* pi)
+{ 
+    int   v;
+    
+	v = i2c_devname_read_byte("CPLD", 0X07);
+	
+    pi->cpld_versions = aim_fstrdup("%d", v);
+
+    return 0;
+}
+
+int
+onlp_sysi_onie_data_get(uint8_t** data, int* size)
+{ 	
+    uint8_t* rdata = aim_zmalloc(256);
+	int fd,rc_size;
+	char  fullpath[20] = {0};
+	
+	sprintf(fullpath, "/dev/mtd7");
+	
+	fd=open(fullpath,O_RDWR);
+	
+	if(fd<0){
+		aim_free(rdata);
+		return ONLP_STATUS_E_INTERNAL ;
+	}
+		
+	rc_size=read(fd,rdata,256);
+	
+	if(rc_size<0||rc_size!=256){
+		aim_free(rdata);
+		return ONLP_STATUS_E_INTERNAL ;
+	}
+			
+    *data = rdata;
+
+    return ONLP_STATUS_OK;
+
+	
+}
+
+void
+onlp_sysi_onie_data_free(uint8_t* data)
+{ 
+    aim_free(data);
+}
+
+
+
+int
+onlp_sysi_oids_get(onlp_oid_t* table, int max)
+{ 
+    int i;
+    onlp_oid_t* e = table;
+    memset(table, 0, max*sizeof(onlp_oid_t));
+
+    /* 1 Thermal sensors on the chassis */
+    for (i = 1; i <= CHASSIS_THERMAL_COUNT; i++) {
+        *e++ = ONLP_THERMAL_ID_CREATE(i);
+    }
+
+    /* LEDs on the chassis */
+    for (i = 1; i <= chassis_led_count(); i++) {
+        *e++ = ONLP_LED_ID_CREATE(i);
+    }
+
+    /* 1 Fans on the chassis */
+    for (i = 1; i <= chassis_fan_count(); i++) {
+        *e++ = ONLP_FAN_ID_CREATE(i);
+    }
+
+    /* 2 PSUs on the chassis */
+    for (i = 1; i <= CHASSIS_PSU_COUNT; i++) {
+        *e++ = ONLP_PSU_ID_CREATE(i);
+    }
+
+    return 0;
+}
+
+int
+onlp_sysi_platform_manage_fans(void)
+{ 
+
+    int rc;
+    onlp_thermal_info_t ti1;
+    onlp_thermal_info_t ti2;
+    int mtemp=0;
+    int new_rpm=0;
+
+    if (chassis_fan_count() == 0) {
+        return ONLP_STATUS_E_UNSUPPORTED;
+    }
+  
+    /* Get temperature */
+    rc = onlp_thermali_info_get(ONLP_THERMAL_ID_CREATE(1), &ti1);
```

Comment:
These must be onlp_thermal_info_get, not onlp_thermali_info_get

Reply:
the function "onlp_thermali_info_get" is our implement ,not the "onlp_thermal_info_get".when we want get the thermal, the the function "onlp_thermali_info_get"  maybe been good,  not the function "onlp_thermal_info_get" . the function "onlp_thermali_info_get" is the underlying  interface, what's more, all the other is realized as the same.

Reply:
This is an unfortunate outcome from my initial choice when naming the interfaces but rest assured that what I am telling you is correct. 

The platform manager that calls these management routines does not hold the API lock. This is the reason you must call the fan_ and thermal_ versions instead of the fani_ and thermali_ versions. If you call the underlying implementation you risk breaking the concurrency protection. 

All other platforms that implement it this way are also implemented incorrectly (though you will only likely see a failure on platforms with userspace implementations). 

Please adjust the code as I have requested.